p2p/auth.py

```diff
@@ -57,8 +57,16 @@
     use_eip8 = False
     initiator = HandshakeInitiator(remote, privkey, use_eip8, token)
     reader, writer = await initiator.connect()
-    aes_secret, mac_secret, egress_mac, ingress_mac = await _handshake(
-        initiator, reader, writer, token)
+    try:
+        aes_secret, mac_secret, egress_mac, ingress_mac = await _handshake(
```

Comment:
I wonder if this could be rewritten using an `asynccontextmanager` so that becomes:

```python

async with initiator.connect() as reader, writer:
    aes_secret, mac_secret, egress_mac, ingress_mac = await _handshake(
            initiator, reader, writer, token)
```

Reply:
In theory yes.  I've thought about this a bit but I'm unsure enough about it that I'd like to punt, with the compromise being adding a larger comment to the section explaining why the solution isn't ideal.